### PR TITLE
Be able to associate files at creation in all entities (#issue/2561)

### LINKED ISSUE #2561: Be able to associate files at creation in all entities
## Use case

- [x] Be able to associate files at creation in all entities, the same it is done for "Cases".
image

- [x] Ensure that the tab to handle files is also available (or any way to manage the uploaded files)

### PR DESCRIPTION
**Pull request by SarahBocognano**
### Proposed changes

* Add "Associated File" to all creation form of entities

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/2561

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Archidoit** [comment]
Some generated files are not included in your commit and are created when we run the back/front:
image

**Comment by Archidoit** [comment]
Create an external reference with an associated file.
image

An error is raised:
image


@Archidoit  The error is actually not due to this issue, it's a bug related to source_name and/or url so I've created an issue to fix it : https://github.com/OpenCTI-Platform/opencti/issues/3386

File: opencti-platform/opencti-front/src/private/components/analysis/external_references/ExternalReferenceCreation.tsx [hunk 1]
```diff
@@ -309,17 +307,7 @@ ExternalReferenceCreationProps
...
-                        classes: { fullWidth: true, variant: 'standard' },
-                        onChange: (event: React.ChangeEvent<HTMLInputElement>) => {
-                          const fileName = event.target.value.split('\\').pop();
-                          const externalIdValue = (document.getElementById('external_id') as HTMLInputElement).value;
-                          if (!externalIdValue && fileName) {
-                            setFieldValue('file', event.currentTarget.files?.[0]);
-                            setFieldValue('external_id', fileName);
-                          }
-                        },
-                      }}
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-front/src/private/components/analysis/external_references/ExternalReferenceCreation.tsx]: Could you explain why all this code was removed ? What was it used for ?

Comment (SarahBocognano) [reply opencti-platform/opencti-front/src/private/components/analysis/external_references/ExternalReferenceCreation.tsx]: It was used to set an external_id with the name of the file if the external_id is not provided by the user. The problem is : 
The code is working fine if you don't put an external_id but not working if you put one.

**Comment by Archidoit** [comment]
Create an attack pattern with an associated file.
Where do you see the file in Attack Pattern since there is no 'Data' tab for this entity type?

Same problem for all the entity types without Data panel (ex: vulnerabilities...).

@Archidoit Those pages were add in my last commit yesterday :)

**Comment by SarahBocognano** [comment]
Update: 
Every comments were fix.

File: opencti-platform/opencti-front/src/private/components/observations/indicators/IndicatorCreation.tsx [hunk 2]
```diff
@@ -348,19 +410,18 @@ export const IndicatorCreationForm = ({
...
   const handleOpen = () => setOpen(true);
   const handleClose = () => setOpen(false);
   const onReset = () => handleClose();
 
-  const paginationOptionsUpdater = R.dissoc('indicatorTypes', R.dissoc('observableTypes', paginationOptions));
-  const updater = (store) => insertNode(
+  const updater = (store: RecordSourceSelectorProxy) => insertNode(
     store,
     'Pagination_indicators',
-    paginationOptionsUpdater,
```
Comment (richard-julien) [inline opencti-platform/opencti-front/src/private/components/observations/indicators/IndicatorCreation.tsx:363]: Need be tested, difficult to understand this modification

File: opencti-platform/opencti-front/src/private/components/analysis/groupings/GroupingCreation.tsx [hunk 3]
```diff
@@ -114,21 +143,38 @@ export const GroupingCreationForm = ({
...
+    objectMarking: defaultMarkingDefinitions ?? [],
+    objectLabel: [],
+    externalReferences: [],
+    file: undefined,
+  };
+
+  const [commit] = useMutation<GroupingCreationMutation>(groupingMutation);
+
+  const onSubmit: FormikConfig<GroupingAddInput>['onSubmit'] = (values, { setSubmitting, setErrors, resetForm }) => {
+    const finalValues: GroupingCreationMutation$variables['input'] = {
```
Comment (richard-julien) [inline opencti-platform/opencti-front/src/private/components/analysis/groupings/GroupingCreation.tsx]: As you directly create the data shape, maybe we can just named that "input"

File: opencti-platform/opencti-front/src/private/components/analysis/groupings/GroupingCreation.tsx [hunk 4]
```diff
@@ -114,21 +143,38 @@ export const GroupingCreationForm = ({
...
+      name: values.name,
+      description: values.description,
+      context: values.context,
+      confidence: parseInt(String(values.confidence), 10),
+      createdBy: values.createdBy?.value,
+      objectMarking: values.objectMarking.map((v) => v.value),
+      objectLabel: values.objectLabel.map((v) => v.value),
+      externalReferences: values.externalReferences.map(({ value }) => value),
+    };
+    if (values.file) {
```
Comment (richard-julien) [inline opencti-platform/opencti-front/src/private/components/analysis/groupings/GroupingCreation.tsx]: Any specific reason to not directly put file in finalValues?

**Review by richard-julien — APPROVED** [review]

**Pull request merged**